Dockerfile.dev

```diff
@@ -0,0 +1,20 @@
+# Dockerfile.dev is a stripped down version of Dockerfile that is used for development purposes.
+# It minimizes number of installs and builds to speed up the development process.
+# Run this on local host only.
+# Generated image is larger than production image. Do not use this for production.
+
+FROM --platform=${BUILDPLATFORM:-amd64} node:22-slim AS build_dev
+
+WORKDIR /usr/app
+
+RUN apt-get update && apt-get install -y g++ make python3 python3-setuptools
+
+COPY . .
+
+RUN yarn install --non-interactive && yarn build
+
+RUN cd node_modules/classic-level && yarn rebuild
```

Comment:
do we really need to do this? we are not building a multiarch image here, so it might be unnecessary

Reply:
> do we really need to do this? we are not building a multiarch image here, so it might be unnecessary

I think we do but maybe @wemeetagain can clarify this. This line is added in #5015 which says it's needed because prebuilt LevelDB does not work well on arm64.

I myself and other team members have arm64 machines and will build and run this locally.

Reply:
I don't think this is an issue for the specific use case we have here which is to build a local dev image, in any case, if this is still an issue we would notice it during development

I would say if it works on macos without that we should keep it out for now to speed up build time